processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -353,9 +359,17 @@ private void pollOnce(Consumer<String, byte[]> consumer) {
 
         timer = Utils.timer();
         records.forEach(record -> {
+            final Closeable trace = null != tracingProvider ? tracingProvider.startTraceFor(record) : null;
```

Comment:
Isn't better to prepare something like `NoopTracingProvider` rather than null checking `tracingProvider` in many places?